main/ejml-core/src/org/ejml/masks/Mask.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ejml.masks;
+
+import org.ejml.MatrixDimensionException;
+import org.ejml.data.Matrix;
+
+/**
+ * Mask used for specifying which matrix entries should be computed
+ */
+public abstract class Mask {
```

Comment:
Should this implement `Matrix`?

Reply:
you can then remove many of the function below and there are generic unit tests for testing these common functions.

Reply:
I would like to avoid implementing `Matrix` here, as it should only be wrapping an existing matrix but not creating a dedicated matrix.
Also I plan to have a mask variant simply backed by a predicate, f.i., only compute the lower triangle of the matrix.

Having the mask simply as Matrix in the operations like vxm is a viable idea. Other GraphBLAS implementations only create the Mask object internally. But as vectors can be represented as simple primitive arrays, I chose to have the Mask object directly as an input.